### PR TITLE
Rewrite convert.py to JS and add SVG optimization and prettier formatting

### PR DESCRIPTION
**Pull request by jeremywiebe**
## Summary:

The original `math-input` repo contained a Python script to generate React components from SVG icon files. In this PR we:

  * convert it to JS
  * build components that use standard props (not `prop-types`)
  * are covered by TypeScript
  * run the SVG through [`svgo`](https://github.com/svg/svgo)
  * run the generated JS through `prettier`

This PR is followed by #202 which uses the new script to generate new icon components (separated out to reduce the size of this PR). 

Issue: LP-11432

## Test plan:

Run the script and check out the components. Also, check out the changes in this diff.

### CODE REVIEW

**Review by jaredly — APPROVED** [review]
👍️

File: packages/math-input/tools/svg-to-react/convert.ts [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+const renderIndex = (filenameMap: {[symbolName: string]: string}) => {
+    const indexContents = Object.keys(filenameMap)
+        .map(
+            (symbolName) =>
+                `export {default as ${symbolName}} from './${filenameMap[symbolName]}';`,
+        )
+        .join("\n");
+
+    return `/**
+* A directory of autogenerated icon components.
```
Comment (handeyeco) [inline packages/math-input/tools/svg-to-react/convert.ts:72]: nit: could we also note that this is an auto-generated file too?

File: packages/math-input/tools/svg-to-react/convert.ts [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+
+const main = async () => {
+    const inputDirName: string = process.argv[2];
+    const outputDirName: string = process.argv[3];
+
+    if (inputDirName == null) {
+        throw new Error("Missing input dir name");
+    }
+    if (outputDirName == null) {
+        throw new Error("Missing input dir name");
```
Comment (handeyeco) [inline packages/math-input/tools/svg-to-react/convert.ts:96]:
```suggestion
        throw new Error("Missing output dir name");
```

**Review by handeyeco — APPROVED** [review]
Is this the same script that would be handling these things? https://github.com/Khan/perseus/pull/458/files#diff-0d9af21924d3fef5f015416ba0e5a0e8a3cc3dacd92e7fa86d07991076cf227eR238-R240

**Review by kevinbarabash — COMMENTED** [review]
Thanks for porting this to TypeScript.  It would be nice to have some fixture tests for this, but it's not a blocker.

**Comment by jeremywiebe** [comment]
Super stale PR. We don't generate SVG anymore for our new keypad. This original keypad is now deleted so this PR makes no sense.

**Pull request closed without merging**